### PR TITLE
Document the tokeniser

### PR DESCRIPTION
**Pull request by GarkGarcia**
This is a follow up to https://github.com/Mathics3/mathics-scanner/issues/9. I documented everything I could find about the bits of the tokeniser used by core.

### CODE REVIEW

**Review by rocky — APPROVED** [review]
LGTM - thanks.

File: mathics_scanner/tokeniser.py
Comment (GarkGarcia) [inline mathics_scanner/tokeniser.py:343]: I don't think this mechanism should be exposed the public API (and therefore it shouldn't be documented in the docstring). If you think about it, all consumers of this library want is to have a functioning WL tokeniser that they can use as a black-box (that what I think at-least).

This definitively should be documented somewhere though. @rocky I'd apprciate if we could merge https://github.com/Mathics3/mathics-scanner/pull/8 before this, so that I can move this information to `implementation.rst`. I also plan to convert `implementation.rst` and the rest of the documentation to a proper Sphinx document before we release the library (we should be pretty easy to do, so it's not gonna take too much time).

File: mathics_scanner/errors.py
Comment (GarkGarcia) [reply mathics_scanner/errors.py]: Fixed on https://github.com/Mathics3/mathics-scanner/pull/11/commits/03ea0c604fedee428a624d52cbe926ab350aa336

**Review by rocky — APPROVED** [review]
I have only a little time to make  a release so we'll merge this in and continue after 1.0.0

**Comment by GarkGarcia** [comment]
@rocky Please wait for me to finish document the tokeniser before the release. It's going to be very fast and it won't break anything since we're only changing the documentation. I'd be very disappointed if you went through with the release without giving me time to fix the documentation, since I've been waiting the entire week for you to merge https://github.com/Mathics3/mathics-scanner/pull/8 so that I could proceed with the changes in here.

**Comment by rocky** [comment]
A release is already done. Sorry. We can make another realase. 

I don't understand why fixing documentation couldn't have been done previously.

**Comment by GarkGarcia** [comment]
> I don't understand why fixing documentation couldn't have been done previously.

I needed `implementation.rst` to be around, and it was added in #8, as I explained in here:

> This definitively should be documented somewhere though. @rocky I'd apprciate if we could merge #8 before this, so that I can move this information to implementation.rst.

Since you didn't reply to my comment, I assumed you were going to wait for me to make the changes.

**Comment by rocky** [comment]
`implementation.rst` is there in master. Just keep committing to it. 

Sorry about the release.

**Comment by mmatera** [comment]
Rocky, please, before de release check that mathics master pass the CI

**Comment by rocky** [comment]
@mmatera Have been working on that right now. Looks like there was longstanding breakage I am finding out about now.  See https://github.com/mathics/Mathics/pull/1146

The failing PossbleQ tests I think are related to something else.

**Pull request merged**
